### PR TITLE
ARROW-2378: [Rust] Rustfmt

### PR DESCRIPTION
**Pull request by max-sixty**
As discussed in https://github.com/apache/arrow/pull/1818

Probably worth merging fairly soon to minimize conflicts. I'll iterate quickly on any feedback.

### CODE REVIEW

**Comment by andygrove** [comment]
Looks good! you have the wrong JIRA number in the title though .. should be 2378

**Review by andygrove — APPROVED** [review]
I'm not authorized to approve PRs officially but I vote that we approve this PR

**Comment by maxim-lian** [comment]
Rust tests pass, though looks like some unrelated failures: https://travis-ci.org/apache/arrow/jobs/361443263#L3527

**Comment by andygrove** [comment]
I wonder if you need to merge laster from upstream master - I'm hitting the same failures in my PRs

**Comment by maxim-lian** [comment]
The branch is up to date on master - I think master is failing; https://travis-ci.org/apache/arrow/builds/361256815

File: ci/travis_script_rust.sh [hunk 1]
```diff
@@ -23,6 +23,9 @@ RUST_DIR=${TRAVIS_BUILD_DIR}/rust
 
 pushd $RUST_DIR
 
+rustup component add rustfmt-preview
+cargo fmt --all -- --write-mode=diff
```
Comment (xhochy) [inline ci/travis_script_rust.sh]: Does this fail if there is a diff?

**Review by xhochy — COMMENTED** [review]
+1, LGTM

Left a minor comment but merging to resolve the potential merge conflicts

**Comment by xhochy** [comment]
@maxim-lian What is your correct JIRA id on https://issues.apache.org/jira/projects/ARROW/summary so that we can assign the resolved ticket to you?

File: ci/travis_script_rust.sh (hunk 1 above)
Comment (maxim-lian) [reply ci/travis_script_rust.sh]: Yes

**Comment by maxim-lian** [comment]
@xhochy I have an account `maximilianroos` on JIRA

**Pull request closed without merging**